src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -259,6 +273,31 @@ private sealed class CachedState
             return factory.IndicateSuccess(msbuildSdkDir, netcoreSdkVersion, propertiesToAdd, itemsToAdd, warnings);
         }
 
+        private static string? GetMSbuildRuntimeVersion(string sdkDirectory, string dotnetRoot)
+        {
+            // 1. Get the runtime version from the MSBuild.runtimeconfig.json file
+            string runtimeConfigPath = Path.Combine(sdkDirectory, "MSBuild.runtimeconfig.json");
+            if (File.Exists(runtimeConfigPath))
+            {
+                var runtimeConfigJson = JsonNode.Parse(File.ReadAllText(runtimeConfigPath)) as JsonObject;
```

Comment:
Is there a more efficient way to read these few fields out of the stream than "fully read it into memory then fully deserialize it"? Probably not _super_ important here but this is allocations in MSBuild.exe . . .

Reply:
the file `MSBuild.runtimeconfig.json` is very short. Example from the built dotnet artifacts:
```json
{
  "runtimeOptions": {
    "tfm": "net10.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "10.0.0-alpha.1.24610.2"
    },
    "configProperties": {
      "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization": false
    }
  }
}
```

Is it a constant thing or the file can be bigger?

Reply:
It _can_ get bigger, but it's going to say on this same order of magnitude. Compare this to the one for .NET 9, for example:

```json
 cat 'C:\Program Files\dotnet\sdk\9.0.101\MSBuild.runtimeconfig.json'
{
  "runtimeOptions": {
    "tfm": "net9.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "9.0.0"
    },
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false,
      "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization": false
    }
  }
}
```

The main thing that will change across releases is what `configProperties` that will be present - and in general that should be quite small.

Reply:
I decided to go the safer route and read json from a stream